src/app/clusters/valve-configuration-and-control-server/valve-configuration-and-control-server-disco.cpp

```diff
@@ -0,0 +1,218 @@
+/**
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ *
+ */
+
+#include "valve-configuration-and-control-server-disco.h"
+
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app-common/zap-generated/ids/Attributes.h>
+#include <app-common/zap-generated/ids/Clusters.h>
+#include <app/AttributeAccessInterface.h>
+#include <app/AttributeAccessInterfaceRegistry.h>
+#include <app/CommandHandlerInterface.h>
+#include <app/CommandHandlerInterfaceRegistry.h>
+#include <app/ConcreteCommandPath.h>
+#include <app/clusters/valve-configuration-and-control-server/valve-configuration-and-control-cluster-logic.h>
+#include <app/data-model/Encode.h>
+#include <app/util/config.h>
+#include <lib/core/CHIPError.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace ValveConfigurationAndControl {
+
+using namespace Attributes;
+using namespace Commands;
+using namespace Protocols::InteractionModel;
+namespace {
+
+CHIP_ERROR TranslateErrorToIMStatus(CHIP_ERROR err)
+{
+    if (err == CHIP_NO_ERROR)
+    {
+        return err;
```

Comment:
This doesn't feel like it belongs in this cluster.

Reply:
We may want to eventually use https://github.com/project-chip/connectedhomeip/blob/master/src/app/data-model-provider/ActionReturnStatus.h#L44

Beyond that though, translating a CHIP_ERROR to another CHIP_ERROR feels odd.